src/Azure.Functions.Cli/Actions/HostActions/StartHostAction.cs

```diff
@@ -480,6 +505,103 @@ public override async Task RunAsync()
             await runTask;
         }
 
+        private async Task<bool> HandleRuntimeParam(bool isCurrentProcessNet6Build, bool isVerbose)
+        {
+            if (string.Equals(SetHostRuntime, "default", StringComparison.OrdinalIgnoreCase))
+            {
+                if (isCurrentProcessNet6Build)
+                {
+                    if (isVerbose)
+                    {
+                        ColoredConsole.WriteLine(VerboseColor("Selected out-of-process host"));
+                    }
+                    await StartHostAsChildProcessAsync(isOutOfProc: true);
+                    return true;
+                }
+            }
+            else if (string.Equals(SetHostRuntime, InProc8HostRuntime, StringComparison.OrdinalIgnoreCase))
+            {
+                if (isCurrentProcessNet6Build && ShouldLaunchInProcNet8AsChildProcess() && await IsInProcNet8Enabled())
+                {
+                    if (isVerbose)
+                    {
+                        ColoredConsole.WriteLine(VerboseColor($"Selected {InProc8HostRuntime} host"));
+                    }
+                    await StartHostAsChildProcessAsync(isOutOfProc: false);
+                    return true;
+                }
+            }
+            else if (string.Equals(SetHostRuntime, InProc6HostRuntime, StringComparison.OrdinalIgnoreCase))
+            {
+                if (isVerbose)
+                {
+                    ColoredConsole.WriteLine(VerboseColor($"Selected {InProc6HostRuntime} host"));
+                }
+                if (!isCurrentProcessNet6Build)
+                {
+                    throw new CliException($"Cannot set host runtime to '{SetHostRuntime}' for the current process. The current process is not a .NET 6 build.");
+                }
+            }
+            else
+            {
+                throw new CliException($"Invalid host runtime '{SetHostRuntime}'. Valid values are 'default', 'in-proc8', 'in-proc6'.");
+            }
+            return false;
+        }
+
+        private async Task<bool> InferHostToLaunchOnStartup(bool isCurrentProcessNet6Build, bool isVerbose)
```

Comment:
Similar comment about this being a bool method and async. Thinking of naming guidelines, what could make more sense for this method name?

- Be Specific About the Boolean Result: The name should help the caller understand what the boolean result represents. For instance, if the method checks if an operation is successful or if a condition is met, reflect that in the name.